feat: test-isolation env var interpolation

davidcavazos opened the pull request:
## Description

- Adds support for environment variable interpolation
- Separates the env var export logic into its own script
- Adds tests for the new script
- Adds a small CLI script for local testing of the env vars
- Adds RUN_ID automatic variable
- Removes the authentication settings from the config

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [x] I have followed guidelines from [CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md) and [Samples Style Guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [ ] **Tests** pass:   `npm test` (see [Testing](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#run-the-tests-for-a-single-sample))
- [x] **Lint** pass:   `npm run lint` (see [Style](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#style))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This pull request is from a branch created directly off of `GoogleCloudPlatform/nodejs-docs-samples`. Not a fork.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new sample directory, and I created [GitHub Actions workflow](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#adding-new-samples) for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

briandorsey reviewed (approved):
Overall LGTM. 

If possible, is there a way we could force a run of testing all the existing packages? It'd be good to make sure the things that have been working still are. (even though we know a few dozen packages need local config)

davidcavazos commented:
The scripts directory was matched as a valid package, so I added it to the ignore list. I removed the workflows directory from the ignore list, the rationale being that changes to the GH actions workflows should trigger all jobs.

Eventually the scripts directory should live in our tools repo and should be tested over there. Ideally we should also add the "run tests via a comment" functionality at some point, but for now this served as a workaround.

davidcavazos commented:
Ignoring the "experimental" failures, we got `generative-ai/snippets`, `healthcare/dicom`, and `media/transcoder` as failures. I know generative-ai is flaky on concurrent runs, the other errors seem like that too. I'm taking notes on those packages and marking them as "flaky", re-running tests.

davidcavazos commented:
`healtcare/dicom` is failing due to some npm install, trying on Node 16.

davidcavazos commented:
`media/transcoder` is getting a timeout, it's taking more than 10 minutes. Updating its setup file.

I also made changes to the config files to trigger all tests.
